framework/Source/BasicOperation.swift

```diff
@@ -95,8 +107,21 @@ open class BasicOperation: ImageProcessingOperation {
             if let alternateRenderingFunction = metalPerformanceShaderPathway, useMetalPerformanceShaders {
                 var rotatedInputTextures: [UInt:Texture]
                 if (firstInputTexture.orientation.rotationNeeded(for:.portrait) != .noRotation) {
-                    let rotationOutputTexture = Texture(device:sharedMetalRenderingDevice.device, orientation: .portrait, width: outputWidth, height: outputHeight)
-                    guard let rotationCommandBuffer = sharedMetalRenderingDevice.commandQueue.makeCommandBuffer() else {return}
+                    guard
+                        let rotationCommandBuffer = sharedMetalRenderingDevice.commandQueue.makeCommandBuffer(),
+                        let rotationOutputTexture = Texture(
+                            device:sharedMetalRenderingDevice.device,
+                            orientation: .portrait,
+                            width: outputWidth,
+                            height: outputHeight
+                        )
+                    else {
+                        assertionFailure("CommandBuffer or Texture creation failed")
+                        removeTransientInputs()
+                        textureInputSemaphore.signal()
```

Comment:
Do we need to call this?

https://github.com/cardinalblue/GPUImage3/pull/22/files#diff-ff11237abc08f188e5a08bf94b2e6d829bf94f01fe50e2daf5a011c809415a72R55-R57

Reply:
Yes we should, and I notice it shouldn't be invoke inside the defer
Let we add and remove them in the next commit, thanks!

Reply:
I'm not really sure if this correct or not. The [L138](https://github.com/cardinalblue/GPUImage3/pull/22/files#diff-ff11237abc08f188e5a08bf94b2e6d829bf94f01fe50e2daf5a011c809415a72R138) call wait.

Reply:
Calling it in defer seems reasonable to me since it calls wait at the beginning.

it only wants to run one at the time.

Reply:
That's weird...sorry I didn't notice this 🤔
I'll check this after I update the implementation for the PR of the mask fask